Mathlib/Topology/Instances/AddCircle/Defs.lean

```diff
@@ -533,15 +533,27 @@ theorem card_addOrderOf_eq_totient {n : ℕ} :
   · rw [← coe_setOf, Nat.card_congr (setAddOrderOfEquiv p hn),
       n.totient_eq_card_lt_and_coprime]
     simp only [Nat.gcd_comm]
+omit [Fact (0 < p)]
+
+theorem finite_torsion {n : ℕ} (hn : 0 < n) : { u : AddCircle p | n • u = 0 }.Finite := by
+  wlog hp : 0 < p generalizing p; swap
+  · have := Fact.mk hp
+    convert Set.finite_range (fun m : Fin n ↦ (↑(↑m / ↑n * p) : AddCircle p))
+    simp_rw [nsmul_eq_zero_iff hn, range, Fin.exists_iff, exists_prop]
+  obtain rfl | hp := eq_or_lt_of_not_gt hp
+  · refine (Set.finite_singleton 0).subset fun u hu ↦ ?_
+    obtain ⟨u, rfl⟩ := QuotientAddGroup.mk_surjective u
+    rw [Set.mem_setOf, ← coe_nsmul] at hu
+    rw [Set.mem_singleton_iff]
+    rw [QuotientAddGroup.eq_zero_iff, zmultiples_zero_eq_bot] at hu ⊢
+    exact (_root_.nsmul_eq_zero_iff hn.ne').mp hu
+  convert ← this (-p) (neg_pos.mpr hp) using 1
+  · rw [AddCircle, zmultiples_neg]
+  exact congr_arg_heq ({u : 𝕜 ⧸ · | n • u = 0}) zmultiples_neg
```

Comment:
For maintainers' consideration, let me link to this comment: https://github.com/leanprover-community/mathlib4/pull/7596#discussion_r2606233907

Reply:
Indeed this is worth noting but I think is OK. I claim that heterogeneous equality arising from a `convert` like this is one of the rare occasions when it can be OK.